core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1395,79 +1389,123 @@ class KafkaApis(val requestChannel: RequestChannel,
       offsetFetchResponse
     }
     requestHelper.sendResponseMaybeThrottle(request, createResponse)
+    CompletableFuture.completedFuture[Unit](())
   }
 
-  private def handleOffsetFetchRequestBetweenV1AndV7(request: RequestChannel.Request): Unit = {
-    val header = request.header
+  private def handleOffsetFetchRequestFromCoordinator(request: RequestChannel.Request): CompletableFuture[Unit] = {
```

Comment:
I have simplified this code even further by pushing the handling of single vs multiple groups in the request/response.

Reply:
Cool. Thanks for simplifying